new/lib/validations.tsx

```diff
@@ -0,0 +1,39 @@
+interface ValidationsMap {
+    [key: string]: (input: any, param?: number) => boolean | string;
+}
+export const validationsMap: ValidationsMap = {
+    //Check required
+    "required": (input: any) => {
+        if (input === undefined || input === null || input === "") {
+            return "is required";
+        }
+        return input?.length > 0;
+    },
+    "min": (input: any, param?: number) => {
+        if (typeof input == "string" && param !== undefined && param !== null) {
+            return (input.length > param);
+        }
+        return `must be at least ${param} long`;
+    },
+    //Check input is String
+    "validateInputIsString": (input: any) => {
+        return (typeof input === "string") ? true : "Invalid Input";
```

Comment:
this should validate the input value not the type, as user cannot alter the object type.

You may have to decide what input is considered as string in your application.